### DISCUSSION TITLE
how can you setcookie after cy.visit?

### DISCUSSION THREAD

**Discussion by muz3** [cypress-io/cypress#18762] [Questions and Help]
our rails backend uses cookiestore generating new cookie on every page load. I would like to see how I can leverage cy.visit to set the new cookie after clicking on <a/> tag ?

have anyone encountered this use case? or maybe there are other alternative ways? any input is much appreciated.

**Comment by jennifer-shehane** [comment]
Cypress runs a real browser behind its own proxy, so it already applies `Set-Cookie` headers automatically, the fresh Rails cookiestore cookie is stored on every cy.visit(), click, or cy.request() without any manual step. 

You can assert it with `cy.getCookie('_yourapp_session')`.

If the real goal is keeping the session across tests (since Cypress clears cookies between tests), use `cy.session()` rather than setting the cookie by hand. For the CSRF flow, see the logging-in__csrf-tokens recipe: https://github.com/cypress-io/cypress-example-recipes/tree/master/examples/logging-in__csrf-tokens
